pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -139,37 +139,80 @@ PyObject	*Float_to_NumpyDouble(float *d)
  * Functions : Numpy array handling functions
  */
 
-void get_strides_and_shape_from_numpy_array(PyObject* arr, int64_t shape[], int64_t strides[], bool c_order)
+/**
+ * Calculate the shapes and strides necessary to pass an array to low-level code.
+ *
+ * If an array is a view on another array then it is passed to low-level code by
+ * saving it into a view with the correct dimensionality and slicing that object.
+ * This function returns the 3 objects necessary for this operation.
+ * Namely:
+ *   - base_shape : The shape of the view with the correct dimensionality.
+ *   - ubounds : The upper bounds of the slice.
+ *   - strides : The strides of the slice.
+ *
+ * E.g.
+ * ```python
+ * a = np.ones((2,3,4,5))
+ * b = a[:,2,::2,:]
+ * ```
+ * Here b has dimensionality 3 so the base_shape, ubounds and strides each contain
+ * 3 elements:
+ *   - base_shape : [4,4,5]
+ *   - ubound : [4,4,5]
+ *   - strides : [3,2,1]
```

Comment:
As discussed in a separate chat, it appears that this example cannot be reproduced by this function in its current form. The result that I obtain is

- `base_shape` : `(2, 6, 10)`
- `ubound` : `(2, 2, 5)`
- `strides` : `(1, 1, 1)`

This should be double checked. If correct, we should either update the function `get_strides_and_shape_from_numpy_array` or its docstring.